src/game_renderer/ViewFrustumCulling.cpp

```diff
@@ -1,75 +1,73 @@
-  planes->push_back(Left / normal_length);
-  Left = Left / normal_length;
-
   Right.x = comp[0][3] - comp[0][0];
-  Right.y = comp[1][3] - comp[1][0];
-  Right.z = comp[2][3] - comp[2][0];
-  Right.w = comp[3][3] - comp[3][0];
-  normal = glm::vec3(Right.x, Right.y, Right.z);
-  normal_length = glm::length(normal);
-  planes->push_back(Right / normal_length);
-  Right = Right / normal_length;
-
   Bottom.x = comp[0][3] + comp[0][1];
-  Bottom.y = comp[1][3] + comp[1][1];
-  Bottom.z = comp[2][3] + comp[2][1];
-  Bottom.w = comp[3][3] + comp[3][1];
-  normal = glm::vec3(Bottom.x, Bottom.y, Bottom.z);
-  normal_length = glm::length(normal);
-  planes->push_back(Bottom / normal_length);
-  Bottom = Bottom / normal_length;
-
   Top.x = comp[0][3] - comp[0][1];
-  Top.y = comp[1][3] - comp[1][1];
-  Top.z = comp[2][3] - comp[2][1];
-  Top.w = comp[3][3] - comp[3][1];
-  normal = glm::vec3(Top.x, Top.y, Top.z);
-  normal_length = glm::length(normal);
-  planes->push_back(Top / normal_length);
-  Top = Top / normal_length;
-
   Near.x = comp[0][3] + comp[0][2];
-  Near.y = comp[1][3] + comp[1][2];
-  Near.z = comp[2][3] + comp[2][2];
-  Near.w = comp[3][3] + comp[3][2];
-  normal = glm::vec3(Near.x, Near.y, Near.z);
-  normal_length = glm::length(normal);
-  Near = Near / normal_length;
-  planes->push_back(Near);
-
   Far.x = comp[0][3] - comp[0][2];
+
+  Left.y = comp[1][3] + comp[1][0];
+  Right.y = comp[1][3] - comp[1][0];
+  Bottom.y = comp[1][3] + comp[1][1];
+  Top.y = comp[1][3] - comp[1][1];
   Far.y = comp[1][3] - comp[1][2];
+  Near.y = comp[1][3] + comp[1][2];
+
+  Left.z = comp[2][3] + comp[2][0];
+  Right.z = comp[2][3] - comp[2][0];
+  Bottom.z = comp[2][3] + comp[2][1];
+  Top.z = comp[2][3] - comp[2][1];
+  Near.z = comp[2][3] + comp[2][2];
   Far.z = comp[2][3] - comp[2][2];
+
+  Left.w = comp[3][3] + comp[3][0];
+  Right.w = comp[3][3] - comp[3][0];
+  Bottom.w = comp[3][3] + comp[3][1];
+  Top.w = comp[3][3] - comp[3][1];
+  Near.w = comp[3][3] + comp[3][2];
   Far.w = comp[3][3] - comp[3][2];
-  normal = glm::vec3(Far.x, Far.y, Far.z);
-  normal_length = glm::length(normal);
+
+  normal_length = glm::length(glm::vec3(Left.x, Left.y, Left.z));
+  Left = Left / normal_length;
+  normal_length = glm::length(glm::vec3(Right.x, Right.y, Right.z));
+  Right = Right / normal_length;
+  normal_length = glm::length(glm::vec3(Bottom.x, Bottom.y, Bottom.z));
+  Bottom = Bottom / normal_length;
+  normal_length = glm::length(glm::vec3(Top.x, Top.y, Top.z));
+  Top = Top / normal_length;
+  normal_length = glm::length(glm::vec3(Near.x, Near.y, Near.z));
+  Near = Near / normal_length;
+  normal_length = glm::length(glm::vec3(Far.x, Far.y, Far.z));
   Far = Far / normal_length;
+
+  planes->push_back(Left);
+  planes->push_back(Right);
+  planes->push_back(Bottom);
+  planes->push_back(Top);
+  planes->push_back(Near);
   planes->push_back(Far);
 
+  std::cout << (std::clock() - start) / (double)(CLOCKS_PER_SEC / 1000) <<  std::endl;
```

Comment:
delete this or put it with the other debug info https://github.com/bbeck13/RhythmRunner/blob/master/src/RhythmRunner.cpp#L38